libs/components/indicators/testing/src/icon/icon-harness.ts

```diff
@@ -0,0 +1,80 @@
+import { HarnessPredicate } from '@angular/cdk/testing';
+import { SkyComponentHarness } from '@skyux/core/testing';
+
+import { SkyIconHarnessFilters } from './icon-harness-filters';
+
+/**
+ * Harness for interacting with an icon component in tests.
+ */
+export class SkyIconHarness extends SkyComponentHarness {
+  /**
+   * @internal
+   */
+  public static hostSelector = 'sky-icon';
+
+  #getIcon = this.locatorFor('.sky-icon');
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a
+   * `SkyIconHarness` that meets certain criteria.
+   */
+  public static with(
+    filters: SkyIconHarnessFilters
+  ): HarnessPredicate<SkyIconHarness> {
+    return SkyIconHarness.getDataSkyIdPredicate(filters);
+  }
+
+  /** Gets the icon name */
+  public async getIconName(): Promise<string | undefined> {
+    const iconClass = await (await this.#getIcon()).getAttribute('class');
+    const regex = /^sky-i-|fa-(?!fw|2xs|xs|sm|lg|2xl|[0-9]+x)/;
```

Comment:
Also, this regular expression might be a strong case to create a `data-` attribute, or a `sky-icon-foobar` class so that we don't need to exclude unknown FontAwesome utility classes to find the name. Let's talk offline as a team.

Reply:
from offline team talk: 
A new attribute is unneccessary for the following reasons - 
- we currently working on an older version of font awesome and have no plans to update to a newer version. So there is no chance of new classes being added that could break this regex. 
- We will also as a team not be adding any new classes that would follow the pattern of `fa-`